src/main/java/zd/zero/waifu/motivator/plugin/assets/RemoteAssetDefinitionService.kt

```diff
@@ -9,24 +11,33 @@ abstract class RemoteAssetDefinitionService<T : AssetDefinition, U : Asset>(
 ) {
     private val random = Random(System.currentTimeMillis())
 
-    fun getRandomAssetByCategory(waifuAssetCategory: WaifuAssetCategory): Optional<U> =
-        pickRandomAsset(remoteAssetManager.supplyAssetDefinitions(), waifuAssetCategory)
-            .flatMap { randomRemoteAsset ->
-                remoteAssetManager.resolveAsset(randomRemoteAsset)
-            }
-            .map { it.toOptional() } // todo: replace with or when supporting only JRE 11+
```

Comment:
Lol, I dunno why I unwrapped the optional just to wrap it again. Thanks for fixing that! :100:

Reply:
I was wondering why when I saw it 😅